### PR TITLE
feat(languages): add tinygo support

### PR DESCRIPTION
**Pull request by vdice**
* Adds [tinygo](https://tinygo.org/) language (er, compiler) support

TODO
 - [x] get debug to work (`wasmtime run tinygo.wasm -g` currently fails)
 - [x] test GH action yamlz

### CODE REVIEW

**Comment by vdice** [comment]
Q: Should we just call this `Go` and have `TinyGo` be a dependency/implementation detail (for building binaries compiled for wasm/wasi)?  Or leave it all as `TinyGo`?

**Comment by itowlson** [comment]
@vdice I would stick to TinyGo.  We don't want to set the expectation that users will get the Google Go compiler and ecosystem.

**Comment by vdice** [comment]
Still not sure how to get debug runs via wasmtime working.  Printing the output in case it rings a bell for others:

```
 $ wasmtime run tinygo.wasm -g
Error: failed to run main module `tinygo.wasm`

Caused by:
    0: Debug information error
    1: comp_dir
```

It seems TinyGo uses [GDB for debugging](https://tinygo.org/docs/tutorials/gdb/) but I'm not sure if there's good VS Code launch support for this?  (I spent some time getting gdb set up on my local OS X system, which included some code signing hurdles.. so not sure how user-friendly it'd be)

Any thoughts @itowlson ?

**Comment by itowlson** [comment]
No, sorry.  Have you reached out to the TinyGo folks for their guidance on debugging Wasm builds?

**Comment by vdice** [comment]
Ok, this should be ready for review!  Apologies, to fix up commit signature, I squashed commits.  The changes added since dusting this PR off are: adding Hippo support (following patterns in pre-existing yo wasm projects) and bumping TinyGo version.  Running via wasmtime w/ the debug flag now works!

While this is in review, I'll look into testing the hippo scenario on an example project...

**Comment by vdice** [comment]
Any chance a review might be possible in the near future?  Looks like we have some great improvements that could be added once this initial PR is in place.  See https://github.com/vdice/yo-wasm/pull/1.  Thanks!

File: templates/tinygo/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+A WASM hello-world written in Go and compiled into WASM via TinyGo
+
+You will need:
+* Go (v1.16 or higher): https://golang.org/doc/install
+* TinyGo: https://tinygo.org/getting-started/install
+* wasmtime: `curl https://wasmtime.dev/install.sh -sSf | bash`
+
+To init: `make init`
+* Just needs running once, this sets up Go module for this project
```
Comment (itowlson) [inline templates/tinygo/README.md:11]: possible typo? 
```suggestion
* Just needs running once, this sets up Go modules for this project
```

File: ts/generators/app/languages/tinygo.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { default as chalk } from 'chalk';
+
+import { Errorable } from '../utils/errorable';
+import { Language } from './language';
+
+export const tinygo: Language = {
+  instructions(): ReadonlyArray<string> {
```
Comment (itowlson) [inline ts/generators/app/languages/tinygo.ts:7]: This has slightly changed - `instructions` and some other functions now take a `Formatter` and use its semantic functions (such as `cmd` for shell commands and `instr` for VS Code instructions).  This allows us to generate terminal colouring for printing or Markdown for adding to the README.

See `rust.ts` for an example. And yeah... it looks like Swift hasn't been updated yet - the old way still works but doesn't give us the extra flexibility to persist the instructions to the readme.

**Review by itowlson — APPROVED** [review]
This looks great - thank you!  I have one minor suggestion, but it's not something we're doing consistently yet, so happy to merge this as is and do that as a follow-up if you prefer - let me know your preference.

File: ts/generators/app/languages/tinygo.ts (hunk 2 above)
Comment (vdice) [reply ts/generators/app/languages/tinygo.ts]: I believe this should now be properly updated.  PTAL when convenient.

**Review by itowlson — APPROVED** [review]
Great stuff - thanks!

**Pull request merged**